flink-tests/src/test/java/org/apache/flink/test/streaming/runtime/BroadcastStateITCase.java

```diff
@@ -169,7 +170,10 @@ public void open(Configuration parameters) throws Exception {
 
 		@Override
 		public void processElement(Long value, KeyedReadOnlyContext ctx, Collector<String> out) throws Exception {
-			ctx.timerService().registerEventTimeTimer(timerTimestamp);
+			long currentTime = al.getAndIncrement();
```

Comment:
@aljoscha I don't understand this. Shouldn't a user have an access to the `currentKey` in this method as well? There should be `ctx.getCurrentKey()` method. Otherwise we are forcing users to duplicate key selector code between here and the application definition.